From 9752e7be0c44ce430e5cc58b1a60d937317be6a0 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:43 -0400
Subject: [tools] perf: Separate lbfd check out of NO_DEMANGLE condition

Message-id: <1408462094-14194-175-git-send-email-jolsa@redhat.com>
Patchwork-id: 88159
O-Subject: [PATCH RHEL7.1 BZ1131394 174/325] perf tools: Separate lbfd check out of NO_DEMANGLE condition
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 3e6a147deef93ddbb899cb394d8d44118289e76a
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Oct 10 22:24:00 2013 +0200

description
===========
We fail build with NO_DEMANGLE with missing -lbfd externals error.
The reason is that we now use bfd code in srcline object:
  perf tools: Implement addr2line directly using libbfd

So we need to check/add -lbfd always now.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 29ad7d6..9680424 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -470,6 +470,10 @@ else
   endif
 endif
 
+ifeq ($(feature-libbfd), 1)
+  EXTLIBS += -lbfd
+endif
+
 ifdef NO_DEMANGLE
   CFLAGS += -DNO_DEMANGLE
 else
@@ -477,9 +481,7 @@ else
     EXTLIBS += -liberty
     CFLAGS += -DHAVE_CPLUS_DEMANGLE_SUPPORT
   else
-    ifeq ($(feature-libbfd), 1)
-      EXTLIBS += -lbfd
-    else
+    ifneq ($(feature-libbfd), 1)
       $(feature_check,liberty)
       ifeq ($(feature-liberty), 1)
         EXTLIBS += -lbfd -liberty
@@ -502,6 +504,10 @@ else
   endif
 endif
 
+ifneq ($(filter -lbfd,$(EXTLIBS)),)
+  CFLAGS += -DHAVE_LIBBFD_SUPPORT
+endif
+
 ifndef NO_ON_EXIT
   ifeq ($(feature-on-exit), 1)
     CFLAGS += -DHAVE_ON_EXIT_SUPPORT
@@ -524,10 +530,6 @@ ifndef NO_LIBNUMA
   endif
 endif
 
-ifndef ($(filter -lbfd,$(EXTLIBS)),)
-  CFLAGS += -DHAVE_LIBBFD_SUPPORT
-endif
-
 # Among the variables below, these:
 #   perfexecdir
 #   template_dir
-- 
1.7.1